### PR TITLE
ID-26

### PR DESCRIPTION
**Pull request by rellfy**
Implements server endpoints for requesting and checking SMS verification codes.
This PR adds the following:
- SMS code verification module (dictionary of `phone_number => { code, date }` with expiry time set from `config.json`).
- Secure route `PUT /api/sms` endpoint: requests a new SMS verification code to phone number.
- Secure route `POST /api/sms` endpoint: verifies SMS code with phone number.
- Basic auth for secure routes, reading users from `API_USERS` ENV variable

To test:
1. copy `server/.env.sample` to `server/.env` -- set `MESSAGEBIRD_KEY` to a valid key
2. copy `server/config.sample.json` to `server/config.json`
2. `cd server && yarn install && yarn start`
3. make a `PUT http://localhost:3000/api/sms` request with the JSON body `{ "phone": "<PHONE NUMBER>" }` and using basic auth with login `admin:password` -- expect SMS to be sent with code (also logged to console)
4. make a `POST http://localhost:3000/api/sms` with the JSON body `{ "phone": "<PHONE NUMBER>", "code": "<VERIFICATION CODE>" }` and using basic auth with login `admin:password`  -- expect `response.success` to be true if code is right.

### CODE REVIEW

File: server/src/getAuthMiddleware.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const basicAuth = require("express-basic-auth");
+let middlware = null;
```
Comment (bitcoinbrisbane) [inline server/src/getAuthMiddleware.js:2]: Spelling `middleware`

File: server/src/getAuthMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  const users = {};
+  for (let string of userStrings) {
+    const data = string.split(":");
+    users[data[0]] = data[1];
+  }
+  return users;
+};
+
+const setMiddleware = () => {
+  middlware = basicAuth({
```
Comment (bitcoinbrisbane) [inline server/src/getAuthMiddleware.js:15]: Spelling

File: server/src/getAuthMiddleware.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    challenge: true,
+    users: getUsers(),
+  });
+};
+
+module.exports = () => {
+  if (middlware == null)
+    setMiddleware();
+  return middlware;
```
Comment (bitcoinbrisbane) [inline server/src/getAuthMiddleware.js:24]: Spelling

File: server/src/getAuthMiddleware.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+const setMiddleware = () => {
+  middlware = basicAuth({
+    challenge: true,
+    users: getUsers(),
+  });
+};
+
+module.exports = () => {
+  if (middlware == null)
```
Comment (bitcoinbrisbane) [inline server/src/getAuthMiddleware.js:22]: Spelling

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
Few small changes

**Review by Matt-Tee — APPROVED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]

**Pull request merged**